### PR TITLE
my friend plank and some tools

### PR DESCRIPTION
**Pull request by cornercube**
## About The Pull Request
adds planks, saws, chisels, stone blocks

ghe

from left to right:
rough stone floor (cobblestone): made with two stone 
stone block floor: made with 1 stone block
stone wall: made with 2 stone block, its the usual wall
fancy stone wall: made with 3 stone block, its wall you sometimes see in town
stone brick wall: also made with 3 stone block, also found a round town

the stoneblock floor is more efficient than the regular stone floor, to encourage turning things into blocks

image
left: short planks make the herringbone floor
right: planks make the diagonal-arranged wooden floor
recipes are more efficient than rough log floors, since you get 2 planks per log

to make planks, craft logs using the handsaw
to make stoneblocks, craft stones using a chisel
you can make saws and chisels out of iron from the black smith

the crude saw is worse than the handsaw and is more bulky, but it can be made from stones and bones
## Why It's Good For The Game
requested by optimism
will make carpenters more important

more swag
you get two more stone walls
and 2 more wood floors

## Pre-Merge Checklist

- [x] You tested this on a local server.
- [x] This code did not runtime during testing.
- [x] You documented all of your changes.

### CODE REVIEW

**Comment by Dorodomki** [comment]
please merg

**Comment by cornercube** [comment]
> please merg

almost ready

**Comment by NPC1314** [comment]
Got some conflicts, needs fixing before it can be tested.

**Comment by cornercube** [comment]
> Got some conflicts, needs fixing before it can be tested.

pain
but I think I got all of the conflicts

File: code/modules/roguetown/roguecrafting/structure.dm [hunk 1]
```diff
@@ -70,7 +70,7 @@
 /datum/crafting_recipe/roguetown/structure/dye_bin_luxury
 	name = "luxury dye bin"
 	result = /obj/machinery/dye_bin
-	reqs = list(/obj/item/grown/log/tree/small = 1,
+	reqs = list(/datum/crafting_recipe/roguetown/plank = 2,
```
Comment (BadAtThisGame302) [inline code/modules/roguetown/roguecrafting/structure.dm:73]: Spoiler alert, someone didn't text their work. This is the wooden plank CRAFTING RECIPE not the WOODEN PLANK itself.

File: code/modules/roguetown/roguecrafting/structure.dm [hunk 2]
```diff
@@ -84,7 +84,7 @@
 /datum/crafting_recipe/roguetown/structure/table
 	name = "wooden table"
 	result = /obj/structure/table/wood/crafted
-	reqs = list(/obj/item/grown/log/tree/small = 1)
+	reqs = list(/datum/crafting_recipe/roguetown/plank = 2)
```
Comment (BadAtThisGame302) [inline code/modules/roguetown/roguecrafting/structure.dm:87]: Spoiler alert, someone didn't text their work. This is the wooden plank CRAFTING RECIPE not the WOODEN PLANK itself.

File: code/modules/roguetown/roguecrafting/structure.dm [hunk 3]
```diff
@@ -132,7 +132,7 @@
 /datum/crafting_recipe/roguetown/structure/closet
 	name = "closet"
 	result = /obj/structure/closet/crate/roguecloset/crafted
-	reqs = list(/obj/item/grown/log/tree/small = 2)
+	reqs = list(/datum/crafting_recipe/roguetown/plank = 2)
```
Comment (BadAtThisGame302) [inline code/modules/roguetown/roguecrafting/structure.dm:135]: Spoiler alert, someone didn't text their work. This is the wooden plank CRAFTING RECIPE not the WOODEN PLANK itself.

File: code/modules/roguetown/roguecrafting/structure.dm [hunk 4]
```diff
@@ -145,7 +145,7 @@
 /datum/crafting_recipe/roguetown/structure/coffin
 	name = "wooden coffin"
 	result = /obj/structure/closet/crate/coffin
-	reqs = list(/obj/item/grown/log/tree/small = 2)
+	reqs = list(/datum/crafting_recipe/roguetown/plank=3)
```
Comment (BadAtThisGame302) [inline code/modules/roguetown/roguecrafting/structure.dm:148]: Spoiler alert, someone didn't text their work. This is the wooden plank CRAFTING RECIPE not the WOODEN PLANK itself.

File: code/modules/roguetown/roguecrafting/structure.dm [hunk 5]
```diff
@@ -233,7 +233,7 @@
 /datum/crafting_recipe/roguetown/structure/barrel
 	name = "wooden barrel"
 	result = /obj/structure/fermenting_barrel/crafted
-	reqs = list(/obj/item/grown/log/tree/small = 1)
+	reqs = list(/datum/crafting_recipe/roguetown/plank=2)
```
Comment (BadAtThisGame302) [inline code/modules/roguetown/roguecrafting/structure.dm:236]: Spoiler alert, someone didn't text their work. This is the wooden plank CRAFTING RECIPE not the WOODEN PLANK itself.

File: code/modules/roguetown/roguecrafting/structure.dm [hunk 6]
```diff
@@ -276,7 +276,7 @@
 /datum/crafting_recipe/roguetown/structure/nicebed
 	name = "nice bed"
 	result = /obj/structure/bed/rogue
-	reqs = list(/obj/item/grown/log/tree/small = 1,
+	reqs = list(/datum/crafting_recipe/roguetown/plank=2,
```
Comment (BadAtThisGame302) [inline code/modules/roguetown/roguecrafting/structure.dm:279]: Spoiler alert, someone didn't text their work. This is the wooden plank CRAFTING RECIPE not the WOODEN PLANK itself.

File: code/modules/roguetown/roguecrafting/structure.dm [hunk 7]
```diff
@@ -288,7 +288,7 @@
 /datum/crafting_recipe/roguetown/structure/door
 	name = "wooden door"
 	result = /obj/structure/mineral_door/wood/deadbolt
-	reqs = list(/obj/item/grown/log/tree/small = 2)
+	reqs = list(/datum/crafting_recipe/roguetown/plank=2)
```
Comment (BadAtThisGame302) [inline code/modules/roguetown/roguecrafting/structure.dm:291]: Spoiler alert, someone didn't text their work. This is the wooden plank CRAFTING RECIPE not the WOODEN PLANK itself.

File: code/modules/roguetown/roguecrafting/structure.dm [hunk 8]
```diff
@@ -343,7 +343,7 @@
 /datum/crafting_recipe/roguetown/structure/windowopen
 	name = "openable window"
 	result = /obj/structure/roguewindow/openclose
-	reqs = list(/obj/item/grown/log/tree/small = 1,
+	reqs = list(/datum/crafting_recipe/roguetown/plank = 1,
```
Comment (BadAtThisGame302) [inline code/modules/roguetown/roguecrafting/structure.dm:346]: Spoiler alert, someone didn't text their work. This is the wooden plank CRAFTING RECIPE not the WOODEN PLANK itself.

File: code/modules/roguetown/roguecrafting/structure.dm [hunk 9]
```diff
@@ -334,7 +334,7 @@
 /datum/crafting_recipe/roguetown/structure/window
 	name = "solid window"
 	result = /obj/structure/roguewindow/solid
-	reqs = list(/obj/item/grown/log/tree/small = 1,
+	reqs = list(/datum/crafting_recipe/roguetown/plank = 1,
```
Comment (BadAtThisGame302) [inline code/modules/roguetown/roguecrafting/structure.dm:337]: Spoiler alert, someone didn't text their work. This is the wooden plank CRAFTING RECIPE not the WOODEN PLANK itself.

File: code/modules/roguetown/roguecrafting/turfs.dm [hunk 10]
```diff
@@ -31,7 +31,7 @@
 /datum/crafting_recipe/roguetown/turfs/woodwall
 	name = "wooden wall"
 	result = /turf/closed/wall/mineral/rogue/wood
-	reqs = list(/obj/item/grown/log/tree/small = 2)
+	reqs = list(/datum/crafting_recipe/roguetown/plank = 2)
```
Comment (BadAtThisGame302) [inline code/modules/roguetown/roguecrafting/turfs.dm:34]: Spoiler alert, someone didn't text their work. This is the wooden plank CRAFTING RECIPE not the WOODEN PLANK itself.

File: code/modules/roguetown/roguecrafting/turfs.dm [hunk 11]
```diff
@@ -6,7 +6,7 @@
 /datum/crafting_recipe/roguetown/turfs/woodfloor
 	name = "wooden floor"
 	result = /turf/open/floor/rogue/ruinedwood
-	reqs = list(/obj/item/grown/log/tree/small = 1)
+	reqs = list(/datum/crafting_recipe/roguetown/plank = 1)
```
Comment (BadAtThisGame302) [inline code/modules/roguetown/roguecrafting/turfs.dm:9]: Spoiler alert, someone didn't text their work. This is the wooden plank CRAFTING RECIPE not the WOODEN PLANK itself.

File: code/modules/roguetown/roguecrafting/turfs.dm [hunk 12]
```diff
@@ -45,7 +45,7 @@
 /datum/crafting_recipe/roguetown/turfs/woodwindow
 	name = "wooden murder hole"
 	result = /turf/closed/wall/mineral/rogue/wood/window
-	reqs = list(/obj/item/grown/log/tree/small = 2)
+	reqs = list(/datum/crafting_recipe/roguetown/plank = 2)
```
Comment (BadAtThisGame302) [inline code/modules/roguetown/roguecrafting/turfs.dm:48]: Spoiler alert, someone didn't text their work. This is the wooden plank CRAFTING RECIPE not the WOODEN PLANK itself.

**Review by BadAtThisGame302 — CHANGES_REQUESTED** [review]

**Comment by Dorodomki** [comment]
noticed the mason doesn't spawn with a chisel to make stone bricks, somewhat the bolt door isn't aviable to craft either, no tables or other stuff is aviable to craft

**Comment by Dorodomki** [comment]
also the small wooden planks aren't craftable or stackable

**Comment by cornercube** [comment]
> noticed the mason doesn't spawn with a chisel to make stone bricks, somewhat the bolt door isn't aviable to craft either, no tables or other stuff is aviable to craft

should be fixed now

**Comment by NPC1314** [comment]
Allright, let me know when the new conflict due to the code cleanup is handled, should be minor, so we can do another test of this

**Comment by NPC1314** [comment]
Until you actually test the PR and provide a video of it working this PR is a leper, second time it breaks crafting.

File: code/modules/roguetown/roguejobs/blacksmith/anvil_recipes/tools.dm [hunk 13]
```diff
@@ -199,6 +199,24 @@
...
+	name = "handsaw (+s)"
+	recipe_name = "a handsaw"
+	req_bar = /obj/item/ingot/iron
+	additional_items = list(/obj/item/grown/log/tree/stick)
+	created_item = /obj/item/rogueweapon/handsaw
+	i_type = "Tools"
+
+/datum/anvil_recipe/tools/chisel
+	name = "chisel (+s) x2"
+	recipe_name = " handsaw"
```
Comment (BadAtThisGame302) [inline code/modules/roguetown/roguejobs/blacksmith/anvil_recipes/tools.dm:212]: Wrong recipe name

**Review by BadAtThisGame302 — CHANGES_REQUESTED** [review]
I think adding a stone saw perhaps, to allow for easier access to planks and creating buildings outside of forcing someone to smith even when there might not be one, would be great.

**Comment by NPC1314** [comment]
The core issue is it introduces a new system without proper information and setup to allow players to adapt to it and use it.
It needs a companion manual book or something giving some hints, along with a summary of the recipe changes in the PR description, for starters.

**Comment by cornercube** [comment]
> I think adding a stone saw perhaps, to allow for easier access to planks and creating buildings outside of forcing someone to smith even when there might not be one, would be great.

added a saw made out of stone and bone, only needs a knife to carve it

File: code/modules/roguetown/roguejobs/craftsman/tools.dm [hunk 14]
```diff
@@ -0,0 +1,232 @@
+/obj/item/rogueweapon/handsaw
+	force = 5
+	possible_item_intents = list(/datum/intent/mace/strike/shovel)
+	gripped_intents = null
+	name = "handsaw"
+	desc = "an iron saw for sawing wood."
```
Comment (BadAtThisGame302) [inline code/modules/roguetown/roguejobs/craftsman/tools.dm:6]: Capitalize the An.

File: code/modules/roguetown/roguejobs/craftsman/tools.dm [hunk 15]
```diff
@@ -0,0 +1,232 @@
...
+	slot_flags = ITEM_SLOT_BACK || ITEM_SLOT_HIP
+	swingsound = list('sound/combat/wooshes/blunt/shovel_swing.ogg','sound/combat/wooshes/blunt/shovel_swing2.ogg')
+	drop_sound = 'sound/foley/dropsound/shovel_drop.ogg'
+	smeltresult = /obj/item/ingot/iron
+	associated_skill = /datum/skill/combat/polearms
+	max_blade_int = 50
+
+/obj/item/rogueweapon/handsaw/crude
+	name = "crude sawing tool"
+	desc = "a serrated piece of flint with a bone handle."
```
Comment (BadAtThisGame302) [inline code/modules/roguetown/roguejobs/craftsman/tools.dm:28]: Capitalize the A.

File: code/modules/roguetown/roguejobs/craftsman/tools.dm [hunk 16]
```diff
@@ -0,0 +1,232 @@
...
+				var/obj/item/natural/bundle/plank/B = new(get_turf(user))
+				B.amount = clamp(plankcount, 2, 3)
+				B.update_bundle()
+				plankcount -= clamp(plankcount, 2, 3)
+		for(var/obj/item/natural/plank/F in get_turf(src))
+			qdel(F)
+
+/obj/item/natural/plankshort
+	name = "short wooden plank"
+	desc = "a smooth wooden plank that has been cut short."
```
Comment (BadAtThisGame302) [inline code/modules/roguetown/roguejobs/craftsman/tools.dm:104]: Capitalize the A.

File: code/modules/cargo/packsrogue/tools.dm [hunk 17]
```diff
@@ -101,6 +101,17 @@
...
+/datum/supply_pack/rogue/tools/handsaw
+	name = "handsaw"
+	cost = 12
+	contains = /obj/item/rogueweapon/handsaw
+
+/datum/supply_pack/rogue/tools/chisel
+	name = "chisel"
+	cost = 12
+	contains =  list(/obj/item/rogueweapon/chisel,
+	/obj/item/rogueweapon/chisel)
```
Comment (BadAtThisGame302) [inline code/modules/cargo/packsrogue/tools.dm:113]: Two chisels?

**Review by BadAtThisGame302 — CHANGES_REQUESTED** [review]

File: code/modules/cargo/packsrogue/tools.dm (hunk 17 above)
Comment (cornercube) [reply code/modules/cargo/packsrogue/tools.dm:113]: yeah, they are smaller items compared to the handsaw, so figured you could get more than one like with daggers

**Review by BadAtThisGame302 — COMMENTED** [review]
I think we can TM it for now and see how it goes once again?

**Comment by cornercube** [comment]
can we test this out again?
every new PR thats merged makes it harder to update this

**Comment by NPC1314** [comment]
Which recipes that can be made now without tools now need the processed material?
How do people get the tools for processing them?
Why dye_bin recipe renamed to _bin?
If you got answers we can TM.

**Comment by cornercube** [comment]
> Which recipes that can be made now without tools now need the processed material?
twig floors for wood
rough stone floors for stone

> How do people get the tools for processing them?
blacksmith can make the saw and chisel out of iron and sticks
you can make a crude saw out of bone and stone

> Why dye_bin recipe renamed to _bin?
that was fixed, its dye_bin

**Comment by NPC1314** [comment]
Got issues at the entry level implementation, locking quite a few recipes behind tools you need to scavenge or hunt to make.
Idea of planks and bricks itself is pretty neat.

**Review by NPC1314 — APPROVED** [review]

**Pull request merged**